projects/sport-data-valley/seeds/protocols/training_log_with_invites.rb

```diff
@@ -0,0 +1,34 @@
+default_protocol_duration = 1.day  # evt eerder dynamisch afbreken
+
+pr_name = 'training_log_with_invites'
+protocol = Protocol.find_by(name: pr_name)
+protocol ||= Protocol.new(name: pr_name)
+protocol.duration = default_protocol_duration
+protocol.informed_consent_questionnaire = nil
+protocol.invitation_text = 'Je bent uitgenodigd door je coach om een vragenlijst in te vullen.'
+protocol.save!
+
+bp_name = 'base-platform-subscription-after-training-with-invites'
+bp_push_subscription = protocol.push_subscriptions.find_by(name: bp_name)
+bp_push_subscription ||= protocol.push_subscriptions.build(name: bp_name)
+bp_push_subscription.method = 'POST'
+bp_push_subscription.url = ENV['PUSH_SUBSCRIPTION_URL']
+bp_push_subscription.save!
+
+protocol.save!
+
+name = 'training_log'
+dagboekvragenlijst_id = Questionnaire.find_by(name: name)&.id
+raise "Cannot find questionnaire: #{name}" unless dagboekvragenlijst_id
```

Comment:
bit nitpicky, but should this whole thing be in a transaction? If the chances of this happening are super slim, I wouldnt' worry about it too much

Reply:
I don't think putting it in a transaction adds much because the entire thing is idempotent. So it doesn't matter where or if we fail, we are always in a good state.